util/timeutil/time.go

```diff
@@ -0,0 +1,160 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package timeutil
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// init initializes `locCache`.
+func init() {
+	// We need set LocalStr when it is in testing process.
+	if LocalStr == "" {
+		LocalStr = "System"
+	}
+	LocCache = &locCache{}
+	LocCache.locMap = make(map[string]*time.Location)
+}
+
+// LocCache is a simple cache policy to improve the performance of 'time.LoadLocation'.
+var LocCache *locCache
+
+// LoadSystemTZ loads system timezone from mysql.tidb and returns the value.
+var LoadSystemTZ func() string
+
+// LocalStr is current TiDB's system timezone name.
+var LocalStr string
+var zoneSources = []string{
+	"/usr/share/zoneinfo/",
+	"/usr/share/lib/zoneinfo/",
+	"/usr/lib/locale/TZ/",
+	// this is for macOS
+	"/var/db/timezone/zoneinfo",
+}
+
+// locCache is a simple map with lock. It stores all used timezone during the lifetime of tidb instance.
+// Talked with Golang team about whether they can have some forms of cache policy available for programmer,
+// they suggests that only programmers knows which one is best for their use case.
+// For detail, please refer to: https://github.com/golang/go/issues/26106
+type locCache struct {
+	sync.RWMutex
+	// locMap stores locations used in past and can be retrieved by a timezone's name.
+	locMap map[string]*time.Location
+}
+
+// GetSystemTZ reads system timezone from `TZ`, the path of the soft link of `/etc/localtime`. If both of them are failed, system timezone will be set to `UTC`.
+// It is exported because we need to use it during bootstap stage. And it should be only used at that stage.
+func GetSystemTZ() string {
+	// consult $TZ to find the time zone to use.
+	// no $TZ means use the system default /etc/localtime.
+	// $TZ="" means use UTC.
+	// $TZ="foo" means use /usr/share/zoneinfo/foo.
+	tz, ok := syscall.Getenv("TZ")
+	switch {
+	case !ok:
+		path, err1 := filepath.EvalSymlinks("/etc/localtime")
+		if err1 == nil {
+			name, err2 := getTZNameFromFileName(path)
+			if err2 == nil {
+				return name
+			}
+			log.Errorln(err2)
+		}
+		log.Errorln(err1)
+	case tz != "" && tz != "UTC":
+		for _, source := range zoneSources {
+			if _, err := os.Stat(source + tz); err == nil {
+				return tz
+			}
+		}
+	}
+	return "UTC"
+}
+
+// getTZNameFromFileName gets IANA timezone name from zoneinfo path.
+// TODO It will be refined later. This is just a quick fix.
+func getTZNameFromFileName(path string) (string, error) {
+	// phase1 only support read /etc/localtime which is a softlink to zoneinfo file
+	substr := "zoneinfo"
+	if strings.Contains(path, substr) {
```

Comment:
`strings.Contains()`  equals to  `strings.Index() == -1`
So you don't `Contains` and then `Index`

Reply:
done